contracts/Minter.sol

```diff
@@ -0,0 +1,174 @@
+   * @param newToken Address of the token to mint
+   * @param startBlock Initial lastMint block
+   * @param newAdmin Initial admin address
+   * @param newMaxToPoolPerBlock Maximum amount minted per block
+   */
+  constructor(address newToken, uint256 startBlock, address newAdmin, uint256 newMaxToPoolPerBlock){
+    require(startBlock >= block.number, "SPSMinter: Start block must be above current block");
+    require(newToken != address(0), 'SPSMinter: Token cannot be address 0');
+    require(newAdmin != address(0), 'SPSMinter: Admin cannot be address 0');
+
+    token = IMintable(newToken);
+    lastMintBlock = startBlock;
+    admin = newAdmin;
+    maxToPoolPerBlock = newMaxToPoolPerBlock;
+
+    require(token.decimals() == 18, "SPSMinter: Token doesn't have 18 decimals");
+
+    emit UpdateAdmin(address(0), newAdmin);
+  }
+
+  /**
+   * @notice Mint tokens to all pools, can be called by anyone
+   */
+  function mint() public {
+    require(totalMinted != cap, "SPSMinter: Cap reached");
+    require(block.number > lastMintBlock, "SPSMinter: Mint block not yet reached");
+
+
+    uint256 mintDifference;
+    unchecked {
+      mintDifference = block.number - lastMintBlock;
+    }
+
+    lastMintBlock = block.number;
+
+    for (uint256 i = 0; i < pools.length; i++){
+      uint256 amount = pools[i].amountPerBlock * mintDifference;
+
+      if(totalMinted + amount >= cap){
+        amount = cap - totalMinted;
+      }
+
+      token.mint(pools[i].receiver, amount);
+      totalMinted = totalMinted + amount;
+
+      emit Mint(pools[i].receiver, amount);
+    }
+  }
+
+  /**
+   * @notice Add new pool, can be called by admin
+   * @param newReceiver Address of the receiver
+   * @param newAmount Amount of tokens per block
+   */
+  function addPool(address newReceiver, uint256 newAmount) external onlyAdmin {
+    require(pools.length <= poolsCap, 'SPSMinter: Pools cap reached');
+    require(newAmount <= maxToPoolPerBlock, 'SPSMinter: Maximum amount per block reached');
+    pools.push(Pool(newReceiver, newAmount));
+    emit PoolAdded(newReceiver, newAmount);
+  }
+
+  /**
+   * @notice Update pool, can be called by admin
+   * @param index Index in the array of the pool
+   * @param newReceiver Address of the receiver
+   * @param newAmount Amount of tokens per block
+   */
+  function updatePool(uint256 index, address newReceiver, uint256 newAmount) external onlyAdmin {
+    require(newAmount <= maxToPoolPerBlock, 'SPSMinter: Maximum amount per block reached');
+    mint();
+    pools[index] = Pool(newReceiver, newAmount);
+    emit PoolUpdated(index, newReceiver, newAmount);
+  }
+
+  /**
+   * @notice Remove pool, can be called by admin
+   * @param index Index in the array of the pool
+   */
+  function removePool(uint256 index) external onlyAdmin {
+    mint();
+    emit PoolRemoved(index, pools[index].receiver, pools[index].amountPerBlock);
+
+    pools[index] = pools[pools.length - 1];
```

Comment:
The calculation of this index can be done in an `unchecked` block

Reply:
Resolved: https://github.com/steem-monsters/sps-minter/commit/958fa509ca0dcbe66e51a34ebb2e35a2d92e4b2f

Reply:
Looking at this again, we actually need a require check that the index being removed is valid, that is, less than pools.length.

Reply:
Resolved: https://github.com/steem-monsters/sps-minter/commit/784672ed244c4f53c181201f7c8dafe17e078946